knora/dsplib/utils/xml_upload.py

```diff
@@ -221,6 +222,68 @@ def _parse_xml_file(input_file: str) -> etree.ElementTree:
+        resclass_name_2_type: infos about the resource classes that exist on the DSP server for the current ontology
+        verbose: verbose switch
+
+    Returns:
+        None if everything went well. Raises a BaseError if there is a problem.
+    """
+    if verbose:
+        print("Perform a deep inspection of your XML file...")
+    knora_properties = resclass_name_2_type[resources[0].restype].knora_properties
+
+    for resource in resources:
+
+        # check that the resource type is consistent with the ontology
+        if resource.restype not in resclass_name_2_type:
+            raise BaseError(
+                f"=========================\n"
+                f"ERROR: Resource '{resource.label}' (ID: {resource.id}) has an invalid resource type "
+                f"'{resource.restype}'. Is your syntax correct? Remember the rules:\n"
+                f" - DSP-API internals: <resource restype=\"restype\">         "
+                        f"(will be interpreted as 'knora-api:restype')\n"
+                f" - current ontology:  <resource restype=\":restype\">        "
+                        f"('restype' must be defined in the 'resources' section of your ontology)\n"
+                f" - other ontology:    <resource restype=\"other:restype\">   "
+                        f"(not yet implemented: 'other' must be defined in the same JSON project file than your ontology)"
+            )
+
+        # validate attribute 'creation_date' of <resource>
+        if resource.creation_date:
+            validate_resource_creation_date(resource.creation_date,
+                                            f"The resource '{resource.label}' (ID: {resource.id}) has an invalid "
+                                            f"creation date. Did you perhaps forget the timezone?")
+
+        # check that the property types are consistent with the ontology
+        resource_properties = resclass_name_2_type[resource.restype].properties.keys()
+        for propname in [prop.name for prop in resource.properties]:
+            if propname not in knora_properties and propname not in resource_properties:
+                raise BaseError(
+                    f"=========================\n"
+                    f"ERROR: Resource '{resource.label}' (ID: {resource.id}) has an invalid property '{propname}'. "
+                    f"Is your syntax correct? Remember the rules:\n"
+                    f" - DSP-API internals: <text-prop name=\"propname\">         "
+                            f"(will be interpreted as 'knora-api:propname')\n"
+                    f" - current ontology:  <text-prop name=\":propname\">        "
+                            f"('propname' must be defined in the 'properties' section of your ontology)\n"
+                    f" - other ontology:    <text-prop name=\"other:propname\">   "
+                            f"(not yet implemented: 'other' must be defined in the same JSON project file than your ontology)"
```

Comment:
personally, I'm not a fan of those multiple strings implicitly being concatenated (plus adding all those `\n`). I think this would be more elegant with a triple-quote multi-line string.

Reply:
But then I get a problem with all the pretty-printing-whitespaces. In the Python code, my multi-line string must be indented, but in the Terminal output it shouldn't. Therefore, I have to concatenate single-line strings.